server/src/main/java/org/elasticsearch/node/NodeService.java

```diff
@@ -125,8 +124,7 @@ public NodeInfo info(
         boolean indices
     ) {
         return new NodeInfo(
-            // TODO: revert to Build.current().version() when Kibana is updated
-            Version.CURRENT.toString(),
+            Build.current().version(),
```

Comment:
Note: this is here to see if everything compiles and pass tests, in serverless too. 
I'll revert it and the change will be done by #136853 once this PR is merged, to keep things nicely separated.